### ISSUE TITLE
logLevel in version 5 doesn't filter console logging

### ISSUE THREAD

**Issue by thesunny** [webdriverio/webdriverio#3102]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.0.0-beta.12
 - **Mode:** Standalone
 - **If WDIO Testrunner, running sync/async:** async
 - **Node.js version: 8.11.1
 - **NPM version:** yarn 1.7.0
 - **Browser name and version:** Chrome 70
 - **Platform name and version:** OSX 10.14
 - **Additional wdio packages used (if applicable):** none

**Config of WebdriverIO**
```js
const config = {
  logLevel: "error",
  capabilities: {
    browserName: "chrome",
    chromeOptions: {
      args: ["--window-size-1024,768"],
    },
  },
  baseUrl: "http://192.168.150.234:3000",
}
```

**Describe the bug**
Set the logLevel to `error` but the driver is displaying all log levels to the console (e.g. DEBUG and INFO)

**To Reproduce**

```js
const browser = remote(config)
browser.url('https://www.google.com') // anything should do it
```

**Expected behavior**
Expect to run the commands without logging to the console

**Log**
Nothing specially relevant in the logs except that they are displaying at all when they shouldn't.

**Additional context**

> Please use the correct markdown

**Comment by christian-bromann** [comment]
@thesunny I couldn't reproduce the issue. The only log content I see when switching to error was:

> 2018-12-05T23:54:36.172Z DEBUG wdio-config: wdio-sync found, running tests synchronous

which is a known "issue". Can you provide a sample script and your output?

**Comment by thesunny** [comment]
@christian-bromann 

Sorry, the bug was more complicated than I thought and I needed to do more to reproduce it.

Here's a breakdown:

- If you create a new `remote` session, the `logLevel` works fine.
- If you create a `remote` session and then you `attach` to that session with the `remote`s config within the same running instance of nodejs, everything works fine
- If you create a `remote` session, then you save the config. In a separate nodejs instance, you `attach` to that session, then the `logLevel` doesn't work.

In order to reproduce, I had to create two separate programs which you should run with `babel-node`.

**First program**
```js
import { remote } from "webdriverio"

const config = {
  logLevel: "error",
  capabilities: {
    browserName: "chrome",
    chromeOptions: {
      args: ["--window-size-1024,768"],
    },
  },
}

async function app() {
  const browser = await remote(config)
  await browser.url("https://www.google.com")
  console.log({
    sessionId: browser.sessionId,
    ...browser.options,
  })
}

app()
```

When this program runs, do not close the `chrome` browser since we `attach` to it in the second program.

The program creates a `config` that can be used by `attach`. I had to add the `sessionId` back in as that is required in addition to the `browser.options` for `attach` to work.

I've also done a `browser.url("https://www.google.com")` call in there to show, in the console, that we aren't seeing any of the logging information except from the known debug issue for `wdio-config`.

Now, I've done a `console.log` on the `config`. That needs to be cut and pasted into the second program:

```js
import { attach } from "webdriverio"

// Cut and paste the config from the console.log and replace the value in config below.
// Don't keep any of these values. Use the value that was logged from last program.
const config = {
  sessionId: "b511f759585306b67482423c4c42f66c",
  protocol: "http",
  hostname: "0.0.0.0",
  port: 4444,
  path: "/wd/hub",
  capabilities: {
    // removed for security
  },
  logLevel: "error",
  // removed for security
}

async function app() {
  const browser = await attach(config)
  await browser.url("https://www.google.com")
  console.log({
    sessionId: browser.sessionId,
    ...browser.options,
  })
}

app()
```

When this second program runs, we see the debug info. Here's a sample I'm seeing:

```
2018-12-06T00:42:41.761Z DEBUG wdio-config: wdio-sync not found, running tests asynchronous
2018-12-06T00:42:42.123Z INFO webdriver: COMMAND navigateTo("https://www.google.com")
2018-12-06T00:42:42.124Z INFO webdriver: [POST] http://0.0.0.0:4444/wd/hub/session/b511f759585306b67482423c4c42f66c/url
2018-12-06T00:42:42.124Z INFO webdriver: DATA { url: 'https://www.google.com' }
```

So basically, in the `attach` the `logLevel` is not respected.

**Comment by thesunny** [comment]
@christian-bromann 

Okay, I think I know how to fix it but I'm not confident enough with the codebase yet so maybe I'll just submit the fix here. It's called out with `### This line here is the fix ###`

```js
export default class WebDriver {
    static async newSession (options = {}, modifier, proto = {}, commandWrapper) {
        const params = validateConfig(DEFAULTS, options)
        logger.setLevel('webdriver', params.logLevel)

        const sessionRequest = new WebDriverRequest(
            'POST',
            '/session',
            {
                capabilities: params.capabilities, // W3C compliant
                desiredCapabilities: params.capabilities // JSONWP compliant
            }
        )

        const response = await sessionRequest.makeRequest(params)
        params.requestedCapabilities = params.capabilities
        params.capabilities = response.value.capabilities || response.value
        params.isW3C = isW3CSession(response.value)

        const prototype = Object.assign(getPrototype(params.isW3C), proto)
        const monad = webdriverMonad(params, modifier, prototype)
        return monad(response.value.sessionId || response.sessionId, commandWrapper)
    }

    /**
     * allows user to attach to existing sessions
     */
    static attachToSession (options = {}, modifier, proto = {}, commandWrapper) {
        if (typeof options.sessionId !== 'string') {
            throw new Error('sessionId is required to attach to existing session')
        }

        // ### This line here is the fix###
        // Note: maybe needs a `validateConfig` like in the `remote` method.
        // Not familiar enough with code to know for certain...
        logger.setLevel('webdriver', options.logLevel)

        options.capabilities = options.capabilities || {}
        options.isW3C = options.isW3C || true
        const prototype = Object.assign(getPrototype(options.isW3C), proto)
        const monad = webdriverMonad(options, modifier, prototype)
        return monad(options.sessionId, commandWrapper)
    }
```

**Comment by christian-bromann** [comment]
@thesunny mind proposing a PR?

**Comment by thesunny** [comment]
@christian-bromann I'd be happy to do this but a warning that it would likely be a few weeks as I've got a lot of commitments right now. The code change is easy, but I feel like I'd need to understand WebDriver code better to make sure I'm not breaking it, understand your unit tests, etc. so I'd want to dedicate a bit of time to make sure I'm doing it right.

**Comment by christian-bromann** [comment]
@thesunny could you please review the fix for this in #3116?

**Comment by thesunny** [comment]
@christian-bromann

I'm not too familiar with the code, but I wonder if that needs a condition attached. Something like:

```js
if (options.logLevel != null) {
  logger.setLevel('webdriver', options.logLevel)
}
```

This way if the `options.logLevel` is not present, we don't `logger.setLevel(undefined)`.

**Comment by christian-bromann** [comment]
@thesunny does it throw an error?

**Comment by uchagani** [comment]
@christian-bromann I'm still seeing a similar error.  I even start `wdio` with the `-l error` parameter and the log show that i only want error logs but I get everything.  


```
~/dev/myProject*: ./node_modules/.bin/wdio -l error wdio.conf.js
2019-03-14T14:17:16.810Z DEBUG @wdio/config: @wdio/sync found, running tests synchronous
2019-03-14T14:17:17.285Z DEBUG @wdio/cli:run: Run suite with config wdio.conf.js and params { _: [ 'wdio.conf.js' ],
  l: 'error',
  logLevel: 'error',
  '$0': 'node_modules/.bin/wdio' }

 RUNNING  0-0 in chrome - /test/specs/home_page.spec.js

Stdout:
2019-03-14T14:17:17.311Z DEBUG @wdio/utils:initialiseServices: initialise wdio service "chromedriver"
2019-03-14T14:17:17.341Z INFO @wdio/cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.46.628411 (3324f4c8be9ff2f70a05a30ebc72ffb013e1a71e) on port 9515
Only local connections are allowed.
Please protect ports used by ChromeDriver and related test frameworks to prevent access by malicious code.
2019-03-14T14:17:17.466Z INFO @wdio/local-runner: Start worker 0-0 with arg: -l,error,wdio.conf.js

Test Suites:   0 passed, 1 total (0% completed)
Time:    🕐  1.41s

 RUNNING  0-0 in chrome - /test/specs/home_page.spec.js

Stdout:
2019-03-14T14:17:17.311Z DEBUG @wdio/utils:initialiseServices: initialise wdio service "chromedriver"
2019-03-14T14:17:17.341Z INFO @wdio/cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.46.628411 (3324f4c8be9ff2f70a05a30ebc72ffb013e1a71e) on port 9515
Only local connections are allowed.
Please protect ports used by ChromeDriver and related test frameworks to prevent access by malicious code.
2019-03-14T14:17:17.466Z INFO @wdio/local-runner: Start worker 0-0 with arg: -l,error,wdio.conf.js

Test Suites:   0 passed, 1 total (0% completed)
Time:    🕒  7.73s

 RUNNING  0-0 in chrome - /test/specs/home_page.spec.js

Stdout:
2019-03-14T14:17:17.311Z DEBUG @wdio/utils:initialiseServices: initialise wdio service "chromedriver"
2019-03-14T14:17:17.341Z INFO @wdio/cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.46.628411 (3324f4c8be9ff2f70a05a30ebc72ffb013e1a71e) on port 9515
Only local connections are allowed.
Please protect ports used by ChromeDriver and related test frameworks to prevent access by malicious code.
2019-03-14T14:17:17.466Z INFO @wdio/local-runner: Start worker 0-0 with arg: -l,error,wdio.conf.js

Test Suites:   0 passed, 1 total (0% completed)
Time:    🕕  7.95s

 RUNNING  0-0 in chrome - /test/specs/home_page.spec.js

Stdout:
2019-03-14T14:17:17.311Z DEBUG @wdio/utils:initialiseServices: initialise wdio service "chromedriver"
2019-03-14T14:17:17.341Z INFO @wdio/cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.46.628411 (3324f4c8be9ff2f70a05a30ebc72ffb013e1a71e) on port 9515
Only local connections are allowed.
Please protect ports used by ChromeDriver and related test frameworks to prevent access by malicious code.
2019-03-14T14:17:17.466Z INFO @wdio/local-runner: Start worker 0-0 with arg: -l,error,wdio.conf.js
2019-03-14T14:17:25.322Z DEBUG @wdio/local-runner: Runner 0-0 finished with exit code 0

Test Suites:   0 passed, 1 total (0% completed)
Time:    🕖  8.03s

"dot" Reporter:
.

Stdout:
2019-03-14T14:17:17.311Z DEBUG @wdio/utils:initialiseServices: initialise wdio service "chromedriver"
2019-03-14T14:17:17.341Z INFO @wdio/cli:Launcher: Run onPrepare hook
Starting ChromeDriver 2.46.628411 (3324f4c8be9ff2f70a05a30ebc72ffb013e1a71e) on port 9515
Only local connections are allowed.
Please protect ports used by ChromeDriver and related test frameworks to prevent access by malicious code.
2019-03-14T14:17:17.466Z INFO @wdio/local-runner: Start worker 0-0 with arg: -l,error,wdio.conf.js
2019-03-14T14:17:25.322Z DEBUG @wdio/local-runner: Runner 0-0 finished with exit code 0
2019-03-14T14:17:25.324Z INFO @wdio/cli:Launcher: Run onComplete hook

Test Suites:   1 passed, 1 total (100% completed)
Time:    🕗  8.03s
```

**Comment by uchagani** [comment]
FYI I'm using latest release: `"webdriverio": "^5.7.2"` and `"@wdio/cli": "^5.7.2"`

**Comment by uchagani** [comment]
@christian-bromann just verified against v4.14.3 and this issue doesn't exist.  definitely a v5 issue

**Comment by uchagani** [comment]
@christian-bromann created a new issue for this.  probably should have done that in the first place, my apologies.

New issue: https://github.com/webdriverio/webdriverio/issues/3729
